components/group/groupStorage.ts

```diff
@@ -0,0 +1,24 @@
+import { RegisteredGroup } from '../../lib/dto/GroupCollection';
+
+
+const groupStorageKey = "money_tracker_group_storage"
+export const getLocalStorageRegisteredGroups: () => RegisteredGroup[] = () => {
+    if (typeof window === 'undefined' || window.localStorage === undefined) {
+        return []
+    } else {
+        const currentStorageItemString = localStorage.getItem(groupStorageKey)
+        if (currentStorageItemString === null) {
+            window.localStorage.setItem(groupStorageKey, JSON.stringify([]))
+            return []
+        }
+        return JSON.parse(currentStorageItemString) as RegisteredGroup[]
+    }
+}
+
+export const setLocalStorageRegisteredGroups: (groups: RegisteredGroup[]) => void = (groups: RegisteredGroup[]) => {
+    if (typeof window === 'undefined' || window.localStorage === undefined) {
+        return []
```

Comment:
a `void` function should probably break with simply `return;` instead of `return []`, it is weird that typescript allowed this.